lib/utils.js

```diff
@@ -842,3 +843,25 @@ function idsHaveDuplicates(ids) {
 function isClass(fn) {
   return fn && fn.toString().startsWith('class ');
 }
+
+/**
+ * Accept an element, and attach the __parent property to it, unless no object given, while also
+ * making sure to check for already created properties
+ *
+ * @param {object} element
+ * @param {Model} parent
+ */
+function applyParentProperty(element, parent) {
+  assert.strictEqual(typeof element, 'object', 'Non object element given to assign parent');
```

Comment:
Is it a good idea to throw an exception when `element` is not an object? I am concerned that most of our code is callback-based and thus not ready to handle errors thrown synchronously. If the error is not caught, then we are going to crash the process via an unhandled exception.

Have you considered to print a warning via `console.warn` and return immediately?

Reply:
The functionality `applyParentProperty` is used only on sync calls in the execution flow (no async or callback based fns), and already other parts in the same flow throw errors synchronously (if you look at the lines above or below where the fn is used). The parent caller who handles the failure of the sync calls - initializations, is responsible for catching the error thrown and providing asynchronously

Reply:
Fair enough :+1: